include/ginkgo/core/multigrid/uniform_coarsening.hpp

```diff
@@ -0,0 +1,141 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_MULTIGRID_UNIFORM_COARSENING_HPP_
+#define GKO_PUBLIC_CORE_MULTIGRID_UNIFORM_COARSENING_HPP_
+
+
+#include <vector>
+
+
+#include <ginkgo/core/base/composition.hpp>
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+#include <ginkgo/core/multigrid/multigrid_level.hpp>
+
+namespace gko {
+namespace multigrid {
+
+
+/**
+ * UniformCoarsening is a simple coarse grid generation algorithm. It
+ * selects the coarse matrix from the fine matrix by constant jumps that can be
+ * specified by the user.
+ *
+ * @tparam ValueType  precision of matrix elements
+ * @tparam IndexType  precision of matrix indexes
+ *
+ * @ingroup MultigridLevel
+ * @ingroup Multigrid
+ * @ingroup LinOp
+ */
+template <typename ValueType = default_precision, typename IndexType = int32>
+class UniformCoarsening
```

Comment:
Should this define copy/move assignment/constructors? The default ones are not completely inline with our normal understanding of those.